sklearn/utils/estimator_checks.py

```diff
@@ -1492,6 +1492,21 @@ def check_no_fit_attributes_set_in_init(name, Estimator):
                 'estimator but in the fit method. Attribute {!r} '
                 'was found in estimator {}'.format(attr, name))
 
+    if(Estimator.__name__ not in ["GaussianProcess", "RandomizedLasso",
```

Comment:
Since the second part of the test checks any set in `__init__`, we won't need the first part anymore. I suggest to remove it and rename `check_no_fit_attributes_set_in_init` to `check_no_attributes_set_in_init`. Is that a good idea?